Content.Server/Tools/ToolSystem.EarthDigging.cs

```diff
@@ -0,0 +1,116 @@
+using System.Threading;
+using Robust.Shared.Map;
+using Content.Server.Tools.Components;
+using Content.Shared.Interaction;
+using Content.Shared.Maps;
+using Content.Shared.Tools.Components;
+
+namespace Content.Server.Tools;
+
+public sealed partial class ToolSystem
+{
+    private void InitializeEarthDigging()
+    {
+        SubscribeLocalEvent<EarthDiggingComponent, AfterInteractEvent>(OnEarthDiggingAfterInteract);
+        SubscribeLocalEvent<EarthDiggingComponent, EarthDiggingCompleteEvent>(OnEarthDigComplete);
```

Comment:
For doafter stuff like these it's usually

```SubscribeLocalEvent<EarthDiggingCompleteEvent>```

and then you TryComp for the component in the subscription to it

This way prevents some potential bugs that can happen if the component is changed/added/removed between the doafter starting and finishing or getting cancelled.

Reply:
TilePrying did it this way, so I just followed convention. I can see how that could create a problem though. I would've thought the event wouldn't fire if the component reached those lifestages though.

Reply:
tilepry is really old, what i posted is the current convention

Reply:
I figured it out. It was necessary to remove the `doAfterEventTarget` parameter being supplied to UseTool in order to get it to fire a broadcast event as opposed to a directed event.